alf/algorithms/actor_critic_algorithm.py

```diff
@@ -93,10 +98,8 @@ def predict(self, time_step: ActionTimeStep, state: ActorCriticState):
             state=ActorCriticState(actor=actor_state),
             info=())
 
-    def rollout(self,
-                time_step: ActionTimeStep,
-                state: ActorCriticState,
-                is_train_step=False):
+    def rollout_action_distribution(self, time_step: ActionTimeStep,
+                                    state: ActorCriticState, mode):
```

Comment:
1. since this algorithm can work with on policy and off policy trainers, the mode can be one of [`ON_POLICY_TRAINING`,`OFF_POLICY_TRAINING`] , and for `OFF_POLICY_TRAINING` it's no necessary to compute value info
2. `OnPolicyAlgorithm.train_step` calls this function, and it needs value info, i think it's better to explicit implement `train_step` for on policy algorithm which support off policy training
3.  Its subclass `PPOAlgorithm` is an off policy algorithm, the mode is `ROLLOUT`, and it needs value info ,it's better to overwrite this function

Reply:
We can do that later. Right now we don't train ActorCriticAlgorithm using off-policy.